EditImprovements/Edit.1r.patch

```diff
@@ -0,0 +1,62 @@
+--- Edit.Mod.txt.orig	2022-02-16 19:28:58.000000000 +0100
++++ Edit.Mod.txt	2022-02-17 15:20:45.759185000 +0100
+@@ -10,12 +10,6 @@
+     pat: ARRAY maxlen OF CHAR;
+     d: ARRAY 256 OF INTEGER;
+ 
+-  PROCEDURE Max(i, j: LONGINT): LONGINT;
+-    VAR m: LONGINT;
+-  BEGIN IF i >= j THEN m := i ELSE m := j END ;
+-    RETURN m
+-  END Max;
+-
+   PROCEDURE Open*;
+     VAR T: Texts.Text;
+     S: Texts.Scanner;
+@@ -123,6 +117,19 @@
+     END
+   END ChangeOffset;
+ 
++  PROCEDURE ShowPos(F: TextFrames.Frame; pos: INTEGER);
++    VAR R: Texts.Reader; ch: CHAR; n: INTEGER;
++  BEGIN
++    IF (pos < F.org) OR (TextFrames.Pos(F, F.X + F.W, F.Y) < pos) THEN
++      IF 256 < pos THEN n := pos - 256 ELSE n := 0 END;
++      Texts.OpenReader(R, F.text, n);
++      REPEAT Texts.Read(R, ch);
++        IF ch = 0DX THEN n := Texts.Pos(R) - 1 END
++      UNTIL Texts.Pos(R) = pos;
++      TextFrames.Show(F, n)
++    END
++  END ShowPos;
++
+   PROCEDURE Search*;  (*uses global variables M, pat, d for Boyer-Moore search*)
+     VAR Text: TextFrames.Frame;
+       V: Viewers.Viewer;
+@@ -156,7 +163,7 @@
+       IF Text.hasCar THEN pos := Text.carloc.pos ELSE pos := 0 END;
+       len := Text.text.len;
+       Texts.OpenReader(R, Text.text, pos);
+-      Forward(M, R, buf); pos := pos + M;
++      Forward(M, R, buf); INC(pos, M);
```

Comment:
I will accept this change if you promise you will send me a patch before I will rebase this on your new repo that will probably already include this line :)